### PR TITLE
Show HTTPRoutes references in K8s Gateway details

### LINKED ISSUE #5708: References for K8s Gateways
Epic #5501 

Analysis of the objects in the k8s Gateways to point to other objects. 

image

For a k8s Gateway, it can reference: 

- Services.

A Gateway is autogenerated when a k8s GW is created. The GW will have a reference to the k8s GW. Ex.

`internal.istio.io/parents: Gateway/gw/test.bookinfo`

### PR DESCRIPTION
**Pull request by leandroberetta**
Resolves #5708

### CODE REVIEW

File: business/references/k8s_gateway_references.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	return result
+}
+
+func (g K8sGatewayReferences) getConfigReferences(gw *k8s_networking_v1alpha2.Gateway) []models.IstioReference {
+	result := make([]models.IstioReference, 0)
+
+	for _, rt := range g.K8sHTTPRoutes {
+		if len(rt.Spec.ParentRefs) > 0 {
+			for _, pr := range rt.Spec.ParentRefs {
+				if string(pr.Name) == gw.Name {
```
Comment (hhovsepy) [inline business/references/k8s_gateway_references.go:34]: Here I would suggest to add "&& string(*pr.Kind) == kubernetes.K8sActualGatewayType && string(*pr.Group) == kubernetes.K8sNetworkingGroupVersionV1Beta1.Group" in 'if' statement, as theoretically parentRef could be to not Gateway.

**Comment by hhovsepy** [comment]
@leandroberetta thanks for the PR, only one suggestion is added, otherwise lgtm:
Screenshot from 2023-01-25 19-27-43

File: business/references/k8s_gateway_references.go (hunk 1 above)
Comment (leandroberetta) [reply business/references/k8s_gateway_references.go]: Thanks @hhovsepy, it's added now.

**Review by hhovsepy — APPROVED** [review]

**Review by josunect — APPROVED** [review]
Works as expected: 

image

Code LGTM. 

Thanks, @leandroberetta !

**Comment by matejnesuta** [comment]
Hello. Is this something, that can be tested either with the integration, or the Cypress suite?

**Comment by leandroberetta** [comment]
> Hello. Is this something, that can be tested either with the integration, or the Cypress suite?

Hi, is tested on the integrations side, but it also can be tested in Cypress looking for the HTTP routes listed in the references.

**Pull request merged**